1️⃣ Estrutura do Dataset

In [1]:
import numpy as np
import pandas as pd

In [2]:
path_csv = r'/home/jvdata/Documentos/Case-SEGMEDIC/src/data/prata/Dados_limpos.csv'

- Carregar o data set

In [3]:
df_limpo = pd.read_csv(path_csv, sep=";")

- Exibir as 5 Primeiras Linhas

In [4]:
df_limpo.head(6)

,Data_Atendimento,ID_paciente,Nome_procedimento,Categoria_procedimento,valor_pago,retorno,Unidade,Metodo_pagamento,ID_profissional
0,2024-01-30,104593,Rx Tornozelo Esquerdo,Raio X,49.0,Não,Centro Medico Matriz,PIX,18252
1,2024-01-30,104593,Rx Joelho,Raio X,49.0,Não,Centro Medico Matriz,PIX,18252
2,2024-02-06,166899,Consulta Clinica Geral,Clinica Geral,19.0,Não,Centro Medico Matriz,PIX,18052
3,2024-02-22,188620,Consulta Clinica Geral,Clinica Geral,19.0,Não,Centro Medico Matriz,PIX,18190
4,2024-02-26,45111,Consulta Clinica Geral,Clinica Geral,19.0,Não,Centro Medico Matriz,PIX,18188
5,2024-03-27,18877,Consulta Clinica Geral,Clinica Geral,19.0,Não,Centro Medico Matriz,PIX,17955


- Quantas linhas e colunas existem no arquivo?

In [5]:
df_limpo.shape

(5050, 9)

Linhas = 5050
Colunas = 9

2️⃣ Valor Atendido por Mês

- Qual o valor total atendido por mês?

In [6]:
#houve um erro na mudança de data type da coluna. corrigir repetindo o processo de mudança de data type.

df_limpo['Data_Atendimento'] = pd.to_datetime(df_limpo['Data_Atendimento'])

In [7]:
df_limpo.dtypes

Data_Atendimento          datetime64[us]
ID_paciente                        int64
Nome_procedimento                    str
Categoria_procedimento               str
valor_pago                       float64
retorno                              str
Unidade                              str
Metodo_pagamento                     str
ID_profissional                    int64
dtype: object

In [8]:
valor_mes_agrupado = df_limpo.groupby(pd.Grouper(key='Data_Atendimento', freq='ME'))['valor_pago'].agg(['sum', 'mean'])

ME = Mounth End

In [9]:
valor_mes_agrupado = valor_mes_agrupado.rename(columns={
    'sum': 'Soma_Total', 
    'mean': 'Media_Mensal'
})
valor_mes_agrupado.head(25)

,Soma_Total,Media_Mensal
Data_Atendimento,,
2024-01-31,510263.75,1457.896429
2024-02-29,7544.13,25.924845
2024-03-31,12312.81,27.983659
2024-04-30,7627.93,25.683266
2024-05-31,9444.45,27.139224
2024-06-30,4967.07,31.239434
2024-07-31,5969.44,30.456327
2024-08-31,3869.89,29.768385
2024-09-30,3390.41,38.527386


Soma total:
- Janeiro = 510.263,75
- Fevereiro = 7.544,13
- Março = 12.312,81	
- Abril = 7.627,93	
- Maio = 9.444,45	

3️⃣ Quantidade de Atendimentos por Mês

- Quantos atendimentos ocorreram em cada mês?

In [10]:
atendimento_mes = df_limpo.groupby(pd.Grouper(key='Data_Atendimento', freq='ME'))['ID_paciente'].count()
atendimento_mes.head(25)

Data_Atendimento
2024-01-31    350
2024-02-29    291
2024-03-31    440
2024-04-30    297
2024-05-31    348
2024-06-30    159
2024-07-31    196
2024-08-31    130
2024-09-30     88
2024-10-31    161
2024-11-30    160
2024-12-31     89
2025-01-31    176
2025-02-28     89
2025-03-31     71
2025-04-30    112
2025-05-31     70
2025-06-30    127
2025-07-31    429
2025-08-31    263
2025-09-30    220
2025-10-31    221
2025-11-30    266
2025-12-31    297
Freq: ME, Name: ID_paciente, dtype: int64

4️⃣ Pacientes com Maior Frequência

Quais são os Top 5 pacientes com maior uso/frequência de atendimento?

In [11]:
freq_pacientes = df_limpo.groupby(pd.Grouper(key='Data_Atendimento', freq='ME'))['ID_paciente'].value_counts().reset_index(name='Frequencia')
freq_pacientes_max = freq_pacientes.groupby('Data_Atendimento').head(1)
print(freq_pacientes_max.head(6))

    Data_Atendimento  ID_paciente  Frequencia
0         2024-01-31        48165          23
100       2024-02-29       220156          21
183       2024-03-31       220158          19
289       2024-04-30       189892          33
379       2024-05-31       136247          25
470       2024-06-30       177214          15


utilizei o reset.index() para redefinir o índice do dataframe e adicionei o parâmetro 'name' para adicionar a coluna 'Frequencia'

In [12]:
df_limpo['ID_paciente'].value_counts()

ID_paciente
210842    81
189892    66
6190      65
202708    61
53863     58
          ..
53542      1
71188      1
196378     1
254615     1
241580     1
Name: count, Length: 425, dtype: int64